crates/solvers/src/tests/legacy/jit_order.rs

```diff
@@ -0,0 +1,122 @@
+                            0101010101010101010101010101010101010101010101010101010101010101\
+                            0202020202020202020202020202020202020202020202020202020202020202\
+                            03",
+                        "interactions": {
+                            "pre": [
+                                {
+                                    "target": "0x2222222222222222222222222222222222222222",
+                                    "value": "200",
+                                    "callData": "0xabcd",
+                                }
+                            ]
+                        }
+                    },
+                    "exec_sell_amount": "100",
+                    "exec_buy_amount": "200",
+                }
+            ],
+        }),
+    }])
+    .await;
+    let config = legacy::create_temp_config_file(&legacy_solver);
+
+    let engine =
+        tests::SolverEngine::new("legacy".to_owned(), config.to_str().unwrap().to_owned()).await;
+
+    let solution = engine
+        .solve(json!({
+            "id": null,
+            "tokens": {},
+            "orders": [],
+            "liquidity": [],
+            "effectiveGasPrice": "15000000000",
+            "deadline": "2106-01-01T00:00:00.000Z"
+        }))
+        .await;
+
+    assert_eq!(
+        solution,
+        json!({
+            "prices": {},
+            "trades": [
+                {
+                    "kind": "jit",
+                    "order": {
+                        "sellToken": "0x2222222222222222222222222222222222222222",
+                        "buyToken": "0x3333333333333333333333333333333333333333",
+                        "receiver": "0x4444444444444444444444444444444444444444",
+                        "sellAmount": "100",
+                        "buyAmount": "200",
+                        "validTo": 1000,
+                        "appData": "0x6000000000000000000000000000000000000000000000000000000000000007",
+                        "feeAmount": "321",
+                        "kind": "sell",
+                        "partiallyFillable": true,
+                        "sellTokenBalance": "erc20",
+                        "buyTokenBalance": "erc20",
+                        "signingScheme": "eip712",
+                        "signature": "0x\
+                            0101010101010101010101010101010101010101010101010101010101010101\
+                            0202020202020202020202020202020202020202020202020202020202020202\
+                            03",
+                        "preInteractions": [
+                            {
+                                "target": "0x2222222222222222222222222222222222222222",
+                                // U256::from_str() expects a hex encoded number whereas
+                                // U256::to_string() produces a base 10 encoded number. That's why
+                                // the original `200` gets converted to `512` here.
```

Comment:
Is this expected behaviour? I feel like we use decimal notation for big integers everywhere. Maybe this was an accidental oversight in the original HTTP API? (cc @sunce86)

Reply:
I put it of as expected because I remembered Niksa running into the same thing while writing some test but actually this is indeed weird.
This behavior got introduced with the eth flow. Fortunately our DB currently doesn't hold any interactions where `value != 0` so I think we can change the format without breaking anything: [be54e0e](https://github.com/cowprotocol/services/pull/1228/commits/be54e0e80f25a316318b390fc71bf984d8bfd652)

Reply:
@fedgiac can you confirm that this change is harmless for the ethflow?

Reply:
As you mentioned, EthFlow pre-interactions exclusively have 0-value, so it shouldn't affect them. This was added to the `foreign_liquidity_orders` interface by @sunce86 for a Yearn use-case, so I just wanted to make sure that they aren't using values for JIT orders (i.e. "foreign liquidity orders").